src/main/scala/io/iohk/ethereum/blockchain/sync/RegularSync.scala

```diff
@@ -112,31 +115,28 @@ trait RegularSync {
 
   private def handleBlockBodies(peer: ActorRef, m: Seq[BlockBody]) = {
     if (m.nonEmpty) {
-      val result = headersQueue.zip(m).map { case (h, b) => blockValidator(h, b) }
+      val result = headersQueue.zip(m).map { case (h, b) => validators.blockValidator.validateHeaderAndBody(h, b) }
```

Comment:
I think we are running the same checks within the Ledger as one of the first steps before executing the transactions, so...if most of the blocks are ok we are running this twice most of the times. I won't tag this as a blocker for merging tho.